src/util/translation.h

```diff
@@ -47,13 +49,39 @@ inline bilingual_str operator+(bilingual_str lhs, const bilingual_str& rhs)
     return lhs;
 }
 
+namespace util {
+/**
+ * Translation function.
+ * If no translation function is set, simply return the input.
+ */
+inline std::string translate(const char* lit)
+{
+    return G_TRANSLATION_FUN ? G_TRANSLATION_FUN(lit) : lit;
+}
+
+struct Translatable {
+    const char* const lit;
+    consteval Translatable(const char* str) : lit{str} { assert(lit); }
+    std::string translate() const { return util::translate(lit); }
+    operator bilingual_str() const { return {lit, translate()}; }
+};
+
+template <unsigned num_params>
+struct BilingualFmt {
+    const ConstevalFormatString<num_params> original;
+    consteval BilingualFmt(Translatable o) : original{o.lit} {}
+};
+} // namespace util
+
+consteval auto _(util::Translatable str) { return str; }
+
 /** Mark a bilingual_str as untranslated */
 inline bilingual_str Untranslated(std::string original) { return {original, original}; }
```

Comment:
slightly unrelated: can we use a const reference here instead?
```suggestion
inline bilingual_str Untranslated(const std::string& original) { return {original, original}; }
```

Reply:
Undefined parameter evaluation order would make it unclear which of them should be explicitly moved though?
https://stackoverflow.com/a/2934909

I hope the compiler introduces an implicit move for whichever parameter it deems to be the last use of the variable.